insights.go

```diff
@@ -19,7 +19,11 @@ func unregisterInsights() error {
 func insightsIsRegistered() bool {
 	cmd := exec.Command("/usr/bin/insights-client", "--status")
 
-	_ = cmd.Run()
+	res := cmd.Run()
 
-	return cmd.ProcessState.Success()
+	if res == nil {
```

Comment:
Could we log the error if the returned error from `Run` is not nil? Something like:

```go
if err := cmd.Run(); err != nil {
    log.Errorf("cannot execute insights-client: %v", err)
}
return cmd.ProcessState.Success()
```

Reply:
Sure, good idea.

Reply:
Hmm, when it is not possible to execute insights-client, then the log is printed to `stderr`. Is it expected behavior? Shouldn't `rhc` write logs to some file in e.g. `/var/log/`?

Reply:
Yea, I'm not sure if printing the error would clobber the pretty-printed status output. There is no logging-to-a-file mechanism at all in `rhc` today, so that would have to be added. The program isn't really meant to be a logging-to-a-file type program. It's a front-end wrapper around the registration utilities in RHEL.

If `insights-client` is not present in the path, I think that should be considered a fatal error. The application has a hard execution dependency on `insights-client`.

Reply:
Yeah, printing the error clobber the pretty-printed status output using your proposal.

When `insights-client` is not present, then it is some corner case (e.g. you run built `rhc` on Fedora, where insights-client is not available), but I think that `rhc` should recover from such situation gracefully...

I just commit update. The output looks like this in this case:

```
[jhnidek@thinkpad-t580 rhc]$ ./rhc status
Connection status for thinkpad-t580:

● Connected to Red Hat Subscription Management
● Cannot execute insights-client: fork/exec /usr/bin/insights-client: no such file or directory
● The rhc daemon is inactive

Manage your Red Hat connector systems: https://red.ht/connector
```

Thoughts? Maybe we could emphasis this error more with red/bold text.

Reply:
Oh yea I like this. It doesn't mangle the output but does include the error message.